test/lib/pipeline.test.js

```diff
@@ -775,6 +775,101 @@ describe('Pipeline Model', () => {
         });
     });
 
+    describe('update', () => {
+        it('updates a pipeline with a different scm repository and branch', () => {
+            const expected = {
+                params: {
+                    admins: { d2lam: true },
+                    id: 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c',
+                    scmRepo: {
+                        branch: 'master',
+                        name: 'foo/bar',
+                        url: 'https://github.com/foo/bar/tree/master'
+                    },
+                    scmUri: 'github.com:12345:master'
+                },
+                table: 'pipelines'
+            };
+            const scmRepo = {
+                name: 'foo/bar',
+                branch: 'master',
+                url: 'https://github.com/foo/bar/tree/master'
+            };
+
+            scmMock.decorateUrl.resolves(scmRepo);
+            userFactoryMock.get.withArgs({ username: 'd2lam' }).resolves({
+                unsealToken: sinon.stub().resolves('foo')
+            });
+            datastore.update.resolves({
+                id: 1,
+                scmUri: 'github.com:5678910:master',
+                scmRepo: {
+                    branch: 'master',
+                    name: 'screwdriver-cd/screwdriver',
+                    url: 'https://github.com/screwdriver-cd/screwdriver/tree/master'
+                },
+                createTime: '2017-01-05T00:55:46.775Z',
+                admins: {
+                    d2lam: true
+                }
+            });
+
+            pipeline.scmUri = 'github.com:12345:master';
+            pipeline.admins = {
+                d2lam: true
+            };
+
+            return pipeline.update().then((p) => {
+                assert.calledWith(scmMock.decorateUrl, { scmUri, token: 'foo' });
+                assert.calledWith(datastore.update, expected);
+                assert.ok(p);
+            });
+        });
+
+        it('updates a pipeline without updating scmUri when it has not changes', () => {
+            const expected = {
+                params: {
+                    admins: { d2lam: true },
+                    id: 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c'
+                },
+                table: 'pipelines'
+            };
+            const scmRepo = {
+                branch: 'master',
+                name: 'screwdriver-cd/screwdriver',
+                url: 'https://github.com/screwdriver-cd/screwdriver/tree/master'
+            };
```

Comment:
the `scmRepo` in this test case is the same as the previous one. Can we re-use it?